### PR TITLE
2018.3xrt

### PR DESCRIPTION
**Pull request by IshitaGhosh**
Fixes for Coverity issues

### CODE REVIEW

File: src/runtime_src/xdp/profile/rt_profile_writers.cpp [hunk 1]
```diff
@@ -80,7 +83,7 @@ namespace XCL {
     const int maxLength = 1024;
     char buf[maxLength];
     ssize_t len;
-    if ((len=readlink("/proc/self/exe", buf, maxLength-1)) != -1) {
+    if (((len=readlink("/proc/self/exe", buf, maxLength-1)) != -1) && (len < maxLength)) {
```
Comment (stsoe) [inline src/runtime_src/xdp/profile/rt_profile_writers.cpp]: This does not seem right.   What if len>=maxLength, e.g. the input is truncated, where is the buffer null terminated?

File: src/runtime_src/xdp/profile/rt_profile_writers.cpp [hunk 2]
```diff
@@ -1095,14 +1098,23 @@ namespace XCL {
     if(SummaryFileName != "") {
       assert(!Summary_ofs.is_open());
       SummaryFileName += FileExtension;
-      openStream(Summary_ofs, SummaryFileName);
+      try {
+        openStream(Summary_ofs, SummaryFileName);
+      } catch(std::runtime_error &err) {
```
Comment (stsoe) [inline src/runtime_src/xdp/profile/rt_profile_writers.cpp]: Is it really safe to continue to writeDocumentHeader when an exception was thrown?

File: src/runtime_src/xdp/profile/rt_profile_writers.cpp [hunk 3]
```diff
@@ -1095,14 +1098,23 @@ namespace XCL {
...
 
     if (TimelineFileName != "") {
       assert(!Timeline_ofs.is_open());
       TimelineFileName += FileExtension;
-      openStream(Timeline_ofs, TimelineFileName);
+      try {
+        openStream(Timeline_ofs, TimelineFileName);
+      } catch(std::runtime_error &err) {
+        std::cout << " ERROR : " << err.what() << " SDAccel Timeline Trace." << std::endl;
+      }
```
Comment (stsoe) [inline src/runtime_src/xdp/profile/rt_profile_writers.cpp]: Is it really safe to continue to writeDocumentHeader when an exception was thrown?

File: src/runtime_src/xdp/profile/rt_profile_writers.cpp (hunk 1 above)
Comment (stsoe) [reply src/runtime_src/xdp/profile/rt_profile_writers.cpp]: Ok, I see.  That's a false positive from Coverity then.

File: src/runtime_src/xdp/profile/rt_profile_writers.cpp (hunk 2 above)
Comment (IshitaGhosh) [reply src/runtime_src/xdp/profile/rt_profile_writers.cpp]: Hi Soren,
writeDocumentHeader itself checks whether the passed ofstream is open. WriterI::openStream throws the exception when the ofstream could not be opened. So, handling the exception and generating error message should be sufficient. All the other methods using the ofsteam checks whether it is open.
Thanks!

File: src/runtime_src/xdp/profile/rt_profile_writers.cpp (hunk 3 above)
Comment (IshitaGhosh) [reply src/runtime_src/xdp/profile/rt_profile_writers.cpp]: Hi Soren,
writeDocumentHeader itself checks whether the passed ofstream is open. WriterI::openStream throws the exception when the ofstream could not be opened. So, handling the exception and generating error message should be sufficient. All the other methods using the ofsteam checks whether it is open.
Thanks!

File: src/runtime_src/xdp/profile/rt_profile_writers.cpp (hunk 2 above)
Comment (stsoe) [reply src/runtime_src/xdp/profile/rt_profile_writers.cpp]: Ok, but this seems a bit odd.   Catching is supposed to handle the exception, but it really doesn't, why not just rethrow?

File: src/runtime_src/xdp/profile/rt_profile_writers.cpp (hunk 3 above)
Comment (stsoe) [reply src/runtime_src/xdp/profile/rt_profile_writers.cpp]: Ok, but this seems a bit odd.   Catching is supposed to handle the exception, but it really doesn't, why not just rethrow?

**Comment by sonals** [comment]
ok to test

File: src/runtime_src/xdp/profile/rt_profile_writers.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (IshitaGhosh) [reply src/runtime_src/xdp/profile/rt_profile_writers.cpp]: This change is on 2018.3 branch.

**Pull request merged**